rustls/src/server/tls13.rs

```diff
@@ -581,40 +581,39 @@ impl CompleteClientHelloHandling {
             .collect();
 
         let supported_groups = suites::KeyExchange::supported_groups();
-        let chosen_group = supported_groups
+
+        let chosen_share = supported_groups
             .iter()
-            .find(|group| share_groups.contains(group))
-            .cloned();
-
-        if chosen_group.is_none() {
-            // We don't have a suitable key share.  Choose a suitable group and
-            // send a HelloRetryRequest.
-            let retry_group_maybe = supported_groups
-                .iter()
-                .find(|group| groups_ext.contains(group))
-                .cloned();
-            self.handshake
-                .transcript
-                .add_message(chm);
+            .find(|group| share_groups.contains(group)) // Find the chosen group
```

Comment:
I think you can use `find_map(|group| shares_ext.iter().find(|share| share.group == *group)` here to avoid doing two lookups.

Reply:
Good idea, and change applied. This worked nicely.